packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -669,6 +670,9 @@ export class Bucket extends BucketBase {
     super(scope, id);
 
     const { bucketEncryption, encryptionKey } = this.parseEncryption(props);
+    if (props.bucketName) {
```

Comment:
Include `&& !cdk.Token.unresolved(props.bucketName)`

Reply:
Thanks for pointing this out. I was not aware of the CDK's token feature.

Fixed in updated PR.